spec/services/payment_provider_customers/adyen_service_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PaymentProviderCustomers::AdyenService, type: :service do
+  let(:adyen_service) { described_class.new(adyen_customer) }
+  let(:customer) { create(:customer, organization:) }
+  let(:adyen_provider) { create(:adyen_provider) }
+  let(:organization) { adyen_provider.organization }
+  let(:adyen_client) { instance_double(Adyen::Client) }
+  let(:payment_links_api) { Adyen::PaymentLinksApi.new(adyen_client, 70) }
+  let(:checkout) { Adyen::Checkout.new(adyen_client, 70) }
+  let(:payment_links_response) { generate(:adyen_payment_links_response) }
+
+  let(:adyen_customer) do
+    create(:adyen_customer, customer:, provider_customer_id: nil)
+  end
+
+  describe '#create' do
+    subject { adyen_service.create }
+
+    before do
+      allow(Adyen::Client).to receive(:new).and_return(adyen_client)
+      allow(adyen_client).to receive(:checkout).and_return(checkout)
+      allow(checkout).to receive(:payment_links_api).and_return(payment_links_api)
+      allow(payment_links_api).to receive(:payment_links).and_return(payment_links_response)
+    end
+
+    context 'when customer does not have an adyen customer id yet' do
+      it 'calls adyen api client payment links' do
+        subject
```

Comment:
RSpec/NamedSubject: Name your test subject if you need to reference it explicitly. (https://rspec.rubystyle.guide/#use-subject, https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/NamedSubject)